client/src/components/ProductCard/ProductCard.jsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import style from './styles.css';
+
+export default class ProductCard extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const {
+      id, image, imageMini, isFavorite, name, onSale, price, rating, related_products, salePrice, brand
+    } = props.product;
+
+    // console.log(props);
+    this.state = {
+      id, image, imageMini, isFavorite, name, onSale, price, rating, related_products, salePrice, brand
+    };
+
+    this.toggleFavorite = this.toggleFavorite.bind(this);
+  }
+
+  toggleFavorite(e) {
+    const { updateProduct } = this.props;
+
+    //remove this once API is working
+    this.setState((prevState) => ({ isFavorite: !prevState.isFavorite }));
+
+    //use this when API is working
+    // const updatedProduct = this.state;
+    // updatedProduct.isFavorite = !updatedProduct.isFavorite;
+    // updateProduct(updatedProduct);
+  }
+
+  render() {
+    // const { product } = props;
+    const { isFavorite, brand, name, imageMini } = this.state;
+    return (
+      <div className={style.container}>
+        <div className={style.stage}>
+          <button type="button" className={isFavorite ? `${style.heart} ${style.is_active}` : style.heart} onClick={this.toggleFavorite}>{ }</button>
```

Comment:
Surprised airbnb isn't freaking out about how long this line is, but in terms of readability this is honestly way better